xarray/backends/dask_lock.py

```diff
@@ -0,0 +1,107 @@
+# The code in this module is taken from dask. For reference,
```

Comment:
can we put this in `locks.py` or, do something similar to a one of our compat module (`pdcompat.py`)?

Reply:
I've moved the code into locks.py as suggested by @jhamman, thanks! Left a note and a permalink to the dask source.